Point_set_processing_3/include/CGAL/scanline_orient_normals.h

```diff
@@ -0,0 +1,552 @@
+    {
+      double angle = CGAL_PI * double(get (scan_angle_map, *it)) / 180.;
+      Vector_3 los = direction * std::sin(angle) + vertical * std::cos(angle);
+      lines_of_sight[idx] = los;
+    }
+
+    std::tie (scan_position, solver_success)
+      = estimate_scan_position (begin, end, point_map, lines_of_sight);
+
+#ifdef CGAL_SCANLINE_ORIENT_VERBOSE
+    if (solver_success && scan_position.z() > mean_z)
+      std::cerr << "SOLVED" << std::endl;
+    else if (!solver_success)
+      std::cerr << "FAILED, solver failure" << std::endl;
+    else
+      std::cerr << "FAILED, scanner under scanline" << std::endl;
+#endif
+  }
+
+  std::size_t idx = 0;
+  for (Iterator it = begin; it != end; ++ it, ++ idx)
+  {
+    const Vector_3 los = lines_of_sight[idx];
+    const Vector_3& normal = get (normal_map, *it);
+    if (normal_along_scanline_is_inverted (normal, los))
+      put (normal_map, *it, -normal);
+  }
+}
+
+template <typename Iterator, typename PointMap, typename NormalMap,
+          typename ScanAngleMap>
+void orient_scanline (Iterator begin, Iterator end,
+                      PointMap point_map,
+                      NormalMap normal_map,
+                      ScanAngleMap,
+                      const Tag_true&) // fallback scan angle
+{
+  using Point_3 = typename boost::property_traits<PointMap>::value_type;
+  using Vector_3 = typename boost::property_traits<NormalMap>::value_type;
+
+  Vector_3 direction;
+  Vector_3 vertical;
+  std::tie (direction, vertical)
+    = scanline_base (begin, end, point_map);
+
+  // Estimate scanner position:
+  // average XY-projected point, located above
+  double mean_x = 0.;
+  double mean_y = 0.;
+  double max_z = -(std::numeric_limits<double>::max)();
+  std::size_t nb = 0;
+
+  for (Iterator it = begin; it != end; ++ it)
+  {
+    const Point_3& p = get (point_map, *it);
+    mean_x += p.x();
+    mean_y += p.y();
+    max_z = (std::max(max_z, p.z()));
+    ++ nb;
+  }
+
+  Iterator last = end; -- last;
+  double length = CGAL::approximate_sqrt (CGAL::squared_distance
+                                          (get (point_map, *begin),
+                                           get (point_map, *last)));
+
+  Point_3 scan_position (mean_x / nb, mean_y / nb,
+                         max_z + length * 2);
+
+  for (Iterator it = begin; it != end; ++ it)
+  {
+    Vector_3 line_of_sight (get(point_map, *it), scan_position);
+    const Vector_3& normal = get (normal_map, *it);
+    if (normal_along_scanline_is_inverted (normal, line_of_sight))
+      put (normal_map, *it, -normal);
+  }
+}
+
+} // namespace internal
+
+} // namespace Point_set_processing_3
+/// \endcond
+
+// ----------------------------------------------------------------------------
+// Public section
+// ----------------------------------------------------------------------------
+
+/**
+   \ingroup PkgPointSetProcessing3Algorithms
+
+   Orients the normals of the range of `points` by estimating a line
```

Comment:
```suggestion
   orients the normals of the range of `points` by estimating a line
```

should be lower case for all functions of this package.